src/test/java/co/cask/hydrator/plugin/DateTransformTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.hydrator.common.test.MockEmitter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * This is an example of how you can build unit tests for your transform.
+ */
+public class DateTransformTest {
+  private static final Schema INPUT = Schema.recordOf("input",
+                                                      Schema.Field.of("a", Schema.of(Schema.Type.STRING)));
+  private static final Schema INPUT2 = Schema.recordOf("input",
+                                                       Schema.Field.of("a",
+                                                                       Schema.nullableOf(Schema.of(Schema.Type.LONG))));
+  private static final Schema INPUT3 = Schema.recordOf("input",
+                                                       Schema.Field.of("a", Schema.of(Schema.Type.LONG)),
+                                                       Schema.Field.of("b", Schema.of(Schema.Type.LONG)));
+
+  private static final Schema OUTPUT = Schema.recordOf("output",
+                                                      Schema.Field.of("a", Schema.of(Schema.Type.STRING)),
+                                                      Schema.Field.of("b", Schema.of(Schema.Type.STRING)));
+  private static final Schema OUTPUT2 = Schema.recordOf("output",
```

Comment:
Can we add a test with non String non Long input field for date format, If the plugin is supposed to accept only long or strings, Can we add that validation in the validate method